src/rgw/rgw_iam_policy.cc

```diff
@@ -715,24 +716,38 @@ static optional<Principal> parse_principal(CephContext* cct, TokenID t,
       return Principal::tenant(std::move(a->account));
     }
 
-    static const char rx_str[] = "([^/]*)/(.*)";
-    static const regex rx(rx_str, sizeof(rx_str) - 1,
-			  ECMAScript | optimize);
-    smatch match;
-    if (regex_match(a->resource, match, rx)) {
-      ceph_assert(match.size() == 3);
+    typedef boost::tokenizer<boost::char_separator<char> > tokenizer;
+    boost::char_separator<char> sep("/");
+    tokenizer tokens(a->resource, sep);
 
-      if (match[1] == "user") {
+    auto it1 = tokens.begin();
+
+    int size = distance(tokens.begin(), tokens.end());
+    auto it2 = tokens.begin();
+    while (it2 != tokens.end()) {
+        if (distance(tokens.begin(), it2) == (size - 1)) {
```

Comment:
@cbodley : Thanks for the suggestion. Definitely looks better. I tried with regexes but didnt achieve what I wanted, and Adam didnt want too many string copies, so went along with tokenize(), but i didnt quite achieve the string copy part. As far as I understand, string_view wont create new string copies, but will just have pointers to chars rights?